spmd/tensor/parallel/style.py

```diff
@@ -27,6 +21,29 @@ class ParallelStyle(ABC):
     _prepare_input: Optional[_Prepare_Input_Func_Type]
```

Comment:
should the prepare_input/output be non-optional type? I think every parallel style should have them specified? This also matches with the `__init__` function where every style must pass those two function?

Reply:
Actually that's a good point, let me remove it.

Reply:
NVM has a solution here.

Reply:
I see that we didn't remove the `Optional`, so I assume there're still cases when a parallel style have `_prepare_input` or `_prepare_output` be None? Could you share more about those cases?

Reply:
Because if you see ColwiseParallel, the prepare_output is None.

Reply:
Removed.